### DISCUSSION TITLE
how to add time to watermark?

### DISCUSSION THREAD

**Discussion by xomarko** [danielkrupinski/Osiris#3833] [Q&A] [answered]
can someone teach me how to add time to the watermark? thanks!

**Comment by lstrsrt** [comment, marked as answer]
```
std::time_t time;
std::tm tm;
char tm_str[9];

time = std::time( nullptr );
localtime_s( &tm, &time );
std::strftime( tm_str, sizeof tm_str, "%T", &tm );
```
After this, tm_str will contain the formatted time string. I assume you know where to use it.

**Reply by xomarko** [reply]
thanks but i absolutely have no idea where to put the code

**Reply by lstrsrt** [reply]
The function where the watermark is generated and rendered, Misc::watermark() in Misc.cpp.
Replace
`ImGui::Text("Osiris | %d fps | %d ms", frameRate != 0.0f ? static_cast<int>(1 / frameRate) : 0, GameData::getNetOutgoingLatency());` with
`ImGui::Text("Osiris | %d fps | %d ms | %s", frameRate != 0.0f ? static_cast<int>(1 / frameRate) : 0, GameData::getNetOutgoingLatency(), tm_str);` to actually render the text.
